crates/edr_provider/src/data.rs

```diff
@@ -3041,6 +3088,33 @@ mod tests {
         Ok(())
     }
 
+    #[cfg(feature = "test-remote")]
+    #[test]
+    fn test_local_account_code_forked() -> anyhow::Result<()> {
+        let mut fixture = ProviderTestFixture::new_forked_at(None, Some(22437572))?;
+
+        let account = *fixture
+            .provider_data
+            .local_accounts
+            .keys()
+            .next()
+            .expect("there are local accounts");
+
+        assert_eq!(
+            account,
+            "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266".parse::<Address>()?
+        );
+
+        let last_block_number = fixture.provider_data.last_block_number();
+        let block_spec = BlockSpec::Number(last_block_number);
+
+        let code = fixture.provider_data.get_code(account, Some(&block_spec))?;
+
+        assert_eq!(code, Bytes::new());
```

Comment:
I'd make this a for-loop over all keys of `edr_defaults::SECRET_KEYS` (requiring a modification of the `fixture`) or add a separate test (after abstracting `static LEAKED_ADDRESSES` calculation) to validate that the pre-computed addresses correctly map to the secret keys.

This is currently not programmatically validated and can result in uncaught regressions in the future.